### PR TITLE
rename nanogpt_generate to nanogpt to also support train

### PR DESCRIPTION
**Pull request by xmfan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109746

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

Differential Revision: [D49522940](https://our.internmc.facebook.com/intern/diff/D49522940)

### CODE REVIEW

**Review by msaroufim — APPROVED** [review]

**Comment by xuzhao9** [comment]
If we are renaming this model, should we also remove `nanogpt_generate` at the same time?

**Comment by xmfan** [comment]
I'll include the updated torchbench commit pin from https://github.com/pytorch/benchmark/pull/1911 in this PR to also remove `nanogpt_generate` here

**Comment by xmfan** [comment]
@xmfan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49522940).

**Review by malfet — APPROVED** [review]

**Review by xuzhao9 — APPROVED** [review]

**Comment by xmfan** [comment]
@msaroufim For some reason, stable_diffusion_text_encoder is running twice, passes the first but fails on the second with infra_error

```
stable_diffusion_text_encoder      pass
stable_diffusion_text_encoder      infra_error
stable_diffusion_unet              pass_due_to_skip
```

Also stable_diffusion_unet OOMs during training, so I had to disable it

**Comment by msaroufim** [comment]
Do you what the error message is for the text encoder failure? curious about that

also how was the unet ooming exactly? if it was on accuracy checks then there's an explicit skip for large models that's better than an outright skip

**Comment by xmfan** [comment]
stable_diffusion_text_encoder infra error: https://github.com/pytorch/pytorch/actions/runs/6278447316/job/17052570700
```
loading model: 0it [00:10, ?it/s]
cpu  eval  stable_diffusion_text_encoder      
WARNING:common:fp64 golden ref were not generated for stable_diffusion_text_encoder. Setting accuracy check to cosine
No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda'
pass
TIMING: entire_frame_compile:32.30441 backend_compile:26.65146
STATS: call_* op count: 824 | FakeTensorMode.__torch_dispatch__:50132 | FakeTensor.__torch_dispatch__:4175 | ProxyTorchDispatchMode.__torch_dispatch__:14158
Dynamo produced 1 graphs covering 824 ops with 0 graph breaks (0 unique)
corrupted size vs. prev_size while consolidating
ERROR
```

stable_diffusion_unet OOM was also in the same job as the above log (inductor_torchbench_dynamic_cpu_accuracy). I disabled it for cpu only, it seems to pass in the cuda accuracy test (https://github.com/pytorch/pytorch/actions/runs/6278447316/job/17053309644)

**Comment by xmfan** [comment]
@xmfan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49522940).

**Comment by xmfan** [comment]
@xmfan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49522940).
